pybikes/samba.py

```diff
@@ -32,35 +36,62 @@ def update(self, scraper = None):
         # clean the data up
         html_data = ''.join(html_data).replace('\n', '').replace('\r', '').replace('"', '')
 
-        stations = re.findall(STATIONS_RGX, html_data)
+        stations = re.findall(Samba._STATIONS_RGX, html_data)
 
         self.stations = []
 
         # The regex will also match for a function defined in the html. This
         # function is in the position of the array, and thus the [:-1]
         for station in stations[:-1]:
-            self.stations.append(SambaStation(station.split(',')))
+            data = station.split(',')
+            name, latitude, longitude, bikes, free, online_status, operation_status = data[3], float(data[0]), float(data[1]), int(data[7]), int(data[8]) - int(data[7]), data[5], data[6]
+            extra = {
+                'address': data[9],
+                'uid': int(data[4]),
+                'slots': int(data[8])
+            }
+            self.stations.append(SambaStation(name, latitude, longitude, bikes, free, online_status, operation_status, extra))
 
-class SambaStation(BikeShareStation):
-    def __init__(self, data):
+class SambaNew(BaseSystem):
+    sync = True
+    _STATIONS_RGX = "var\ beaches\ =\ \[(.*?)\,\];"
+
+    def __init__(self, tag, meta, url):
+        super(SambaNew, self).__init__(tag, meta)
+        self.feed_url = url
+
+    def update(self, scraper = None):
+        if scraper is None:
+            scraper = utils.PyBikesScraper()
+        scraper.setUserAgent(USERAGENT)
+
+        html = scraper.request(self.feed_url)
+
+        stations_html = re.findall(SambaNew._STATIONS_RGX, html)
+        stations = ast.literal_eval(stations_html[0])
```

Comment:
I was referring to `ast.literal_eval`. It works because there's an overlap between js and python syntax of declaring lists and dicts, but I don't like it as a tool for parsing js objects (and is usually not recommended).